enterprise/server/cmd/ci_runner/bazel_wrapper/bazel_wrapper.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/creack/pty"
+	"github.com/google/shlex"
+)
+
+// This is a wrapper to Bazel that appends Bazel flags.
+var (
+	BazelWrapperFlagset = flag.NewFlagSet("bazel_wrapper", flag.ContinueOnError)
+
+	rootPath = BazelWrapperFlagset.String("root_path", "", "The absolute path of the workspace root dir. See ci_runner/main.go:workspace for more details.")
+
+	bazelBinary       = BazelWrapperFlagset.String("bazel_bin", "", "The bazel binary to invoke.")
+	bazelStartupFlags = BazelWrapperFlagset.String("bazel_startup_flags", "", "Startup flags to pass to bazel. The value can include spaces and will be properly tokenized.")
+	extraBazelArgs    = BazelWrapperFlagset.String("extra_bazel_args", "", "Extra flags to pass to the bazel command. The value can include spaces and will be properly tokenized.")
+)
+
+const (
+	// buildbuddyBazelrcPath is the path where we write a bazelrc file to be
+	// applied to all bazel commands. The path is relative to the runner workspace
+	// root, which notably is not the same as the bazel workspace / git repo root.
+	//
+	// This bazelrc takes lower precedence than the workspace .bazelrc.
+	buildbuddyBazelrcPath = "buildbuddy.bazelrc"
+	// Name of the bazel output base dir. This is written under the workspace
+	// so that it can be cleaned up when the workspace is cleaned up.
+	outputBaseDirName = "output-base"
+
+	// ANSI codes for cases where the aurora equivalent is not supported by our UI
+	// (ex: aurora's "grayscale" mode results in some ANSI codes that we don't currently
+	// parse correctly).
+	ansiGray  = "\033[90m"
+	ansiReset = "\033[0m"
+)
+
+func main() {
+	args := os.Args[1:]
+	err := BazelWrapperFlagset.Parse(args)
+	if err != nil {
+		log.Fatalf("%s", err.Error())
+	}
+
+	if *bazelBinary == "" {
+		log.Fatalf("--bazel-bin is required.")
+	}
+	args = removeWrapperFlags(args)
+
+	args, err = appendStartupArgs(args)
+	if err != nil {
+		log.Fatalf("%s", err.Error())
+	}
+	args, err = appendBazelArgs(args)
+	if err != nil {
+		log.Fatalf("%s", err.Error())
+	}
+
+	io.WriteString(os.Stderr, ansiGray+fmt.Sprintf("Running command: %s %s\n\n", *bazelBinary, strings.Join(args, " "))+ansiReset+" ")
+	if _, err := runCommand(*bazelBinary, args, os.Stderr); err != nil {
+		log.Fatalf("%s", err.Error())
+	}
```

Comment:
I thought it would be helpful for debugging to still print the actual bazel command we are running. i.e. The ci_runner will print something like "bazel build //..."

The extra log line will print "Running command:  /actual/path/to/bazel --bazelrc=buildbuddy.bazelrc build //...."

Reply:
For debugging, the full command will also be sent up on the BES stream and displayed in the BB UI. I think we should lean towards keeping this wrapper as minimal as possible because it increases the surprise factor a little bit, like "why is there this extra output that I don't see locally, is this some sort of bazel wrapper?" Also if people try to parse bazel's output, this extra output might make that more difficult. I think we should start by making this behave like "just bazel" as much as possible and maybe if people want extra fancy features we can maybe add those behind env vars or something

Reply:
Fair enough - removed